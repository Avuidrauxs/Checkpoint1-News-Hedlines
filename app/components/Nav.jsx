const React = require('react');
const { Link, IndexLink } = require('react-router');

export default class Nav extends React.Component{

  constructor(props){
    super(props);
    this.state ={
      token: false
    }
  }

  checkToken(){
    const tempToken = localStorage.getItem('jwtToken');
    if(tempToken){
      this.setState({
        token: false
      })
    } else {
      this.setState({
        token: true
      })
    }
  }
  logOut() {
    localStorage.clear();
    console.log('cleared');
  }
  onSearch(e) {
    e.preventDefault();
    alert('Not yet wired up!');
  }
  render() {
    return (
      <div className="top-bar">
        <div className="top-bar-left">
          <ul className="menu">
            <li className="menu-text"><h3>Bad News App</h3></li>
          </ul>
        </div>
        <div className="top-bar-right">
          <form>
            <ul className="menu" hidden={this.state.token}>
              <Link to="/" className="button" onClick={this.logOut()}>LOG OUT</Link>
            </ul>
          </form>
        </div>
      </div>
    );
  }
}
