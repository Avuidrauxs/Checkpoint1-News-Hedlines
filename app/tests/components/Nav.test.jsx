import React from 'react';
import renderer from 'react-test-renderer';
import expects from 'expect';
import sinon from 'sinon';
import { mount } from 'enzyme';


import Nav from '../../components/Nav';
const userProfile = {
  googleId: '106092711356966952734',
  imageUrl: 'https://lh6.googleusercontent.com/-V-2u0o-CEFI/AAAAAAAAAAI/AAAAAAAABHo/QQ-SVzoEiag/s96-c/photo.jpg',
  email: 'audax338@gmail.com',
  name: 'Audax Anchirinah',
  givenName: 'Audax'
};
localStorage.setItem('profile', JSON.stringify(userProfile));
describe('Nav', () => {
  it('should exist', () => {
    expects(Nav).toExist();
  });
  it('should render a snapshot', () => {
    const tree = renderer.create(<Nav />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should access class method "handleFavouriteBtn" in props', () => {
    const handleFavouriteBtn = jest.fn();
    const component = mount(<Nav onClick={handleFavouriteBtn} />);
    expect(component.props().onClick).toBe(handleFavouriteBtn);
  });
  it('should not call the handleFavouriteBtn Nav is mounted', () => {
    localStorage.setItem('profile', JSON.stringify(userProfile));
    const wrapper = mount(<Nav />);
    const spy = sinon.stub(Nav.prototype, 'handleFavouriteBtn');
    wrapper.instance().handleFavouriteBtn();
    expect(spy.callCount).toBe(0);
  });
});
