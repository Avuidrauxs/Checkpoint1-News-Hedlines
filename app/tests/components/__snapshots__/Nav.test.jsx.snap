// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render a snapshot 1`] = `
<div
  className="top-bar"
>
  <div
    className="top-bar-left"
  >
    <ul
      className="menu menu-left"
    >
      <li>
        <h4>
          Bad News App
        </h4>
      </li>
    </ul>
  </div>
  <div
    className="profile-pic"
    style={
      Object {
        "marginRight": "30px",
      }
    }
  >
    <img
      alt="no-pic"
      src="https://lh6.googleusercontent.com/-V-2u0o-CEFI/AAAAAAAAAAI/AAAAAAAABHo/QQ-SVzoEiag/s96-c/photo.jpg"
    />
  </div>
  <div
    className="top-bar-right"
  >
    <ul
      className="menu menu-right"
    >
      <li>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "green",
              "marginRight": "10px",
            }
          }
        >
          View Favourites
        </button>
      </li>
      <li>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "red",
              "marginRight": "80px",
            }
          }
        >
          Logout, 
          Audax
        </button>
      </li>
    </ul>
  </div>
</div>
`;
